packages/bitcore-lib-cash/lib/script/interpreter.js

```diff
@@ -420,13 +439,24 @@ Interpreter.prototype.checkDataSignatureEncoding = function(buf) {
  * Translated from bitcoind's CheckPubKeyEncoding
  */
 Interpreter.prototype.checkPubkeyEncoding = function(buf) {
-  if ((this.flags & Interpreter.SCRIPT_VERIFY_STRICTENC) !== 0 && !PublicKey.isValid(buf)) {
+  if ((this.flags & Interpreter.SCRIPT_VERIFY_STRICTENC) !== 0 && !(PublicKey.isValid(buf) || IsCompressedOrUncompressedPubkey(buf))) {
     this.errstr = 'SCRIPT_ERR_PUBKEYTYPE';
     return false;
   }
   return true;
 };
 
+function IsCompressedOrUncompressedPubkey(bufPubkey) {
```

Comment:
maybe this should be named "isPubKey" ?

Reply:
I got the name of this function from this line in the bitcoin-abc repo here https://github.com/Bitcoin-ABC/bitcoin-abc/blob/8f4fbf8de1e10c195f9a4b5dbfd7a084682fd27a/src/script/sigencoding.cpp#L289. Should we still try to change it?